Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1100,15 +1100,15 @@ class BrowserViewController: UIViewController {
             print("Cannot navigate in tab without a webView")
             return
         }
-        
-        webView.evaluateJavaScript("\(ReaderModeNamespace).checkReadability()", completionHandler: nil)
 
         if let url = webView.url {
-            if !url.isInternalScheme, !url.isFileURL {
+            if InternalURL(url) == nil, !url.isFileURL {
```

Comment:
nit: in retrospect, there are a lot of these stand-alone 'isInternalScheme' checks, maybe URL extension for `isInternalScheme` is better.